src/basic/alloc-util.h

```diff
@@ -242,4 +242,18 @@ static inline size_t malloc_sizeof_safe(void **xp) {
                 (char*) memdupa_suffix0(_t, strnlen(_t, (n)));          \
         })
 
+/* Free every element of the array. */
+static inline void free_many(void **p, size_t n) {
+        if (!p)
+                return;
```

Comment:
I think this check can be dropped. We will only evaluate `p` if `n > 0`.

Reply:
Yeah. `assert(p || n == 0)` is better.